### PR TITLE
feat(triggers): introduce triggers parsing utility

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
Needed for tooltips / popovers as described in:
https://github.com/ng-bootstrap/core/issues/8#issuecomment-225168469

### CODE REVIEW

File: src/util/triggers.spec.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    it('should throw when more than one manual trigger detected', () => {
+      expect(() => {
+        parseTriggers('manual click manual');
+      }).toThrow('Triggers parse error: only one manual trigger is allowed');
+    });
+
+    it('should throw when manual trigger is mixed with other triggers', () => {
+      expect(() => {
+        parseTriggers('click manual');
+      }).toThrow('Triggers parse error: manual trigger can\'t be mixed with other triggers');
```
Comment (wesleycho) [inline src/util/triggers.spec.ts]: Could use a template string here to avoid escaping

**Comment by wesleycho** [comment]
Probably should be a test to test it handling strings with newline characters appropriately as well.

**Comment by pkozlowski-opensource** [comment]
> Probably should be a test to test it handling strings with newline characters appropriately as well.

I don't think ng2 allows new lines in attribute values (HTML5 spec probably doesn't allow them either).

**Comment by wesleycho** [comment]
It does - I actually implemented that to fix an issue reported against lack of support against that.

**Comment by pkozlowski-opensource** [comment]
@wesleycho you are right! Will add a test for newline characters case. Thnx!

**Pull request closed without merging**